Jetpack Partner Portal: implement WPCOM atomic hosting notification banner

yashwin opened the pull request:
Closes https://github.com/Automattic/jetpack-avalon/issues/43

## Proposed Changes

This PR implements the WPCOM atomic hosting notification banner.

### Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Run `git checkout add/implement-wpcom-atomic-hosting-notification-banner` and `yarn start-jetpack-cloud`.
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
3. Open `/jetpack-cloud/sections/partner-portal/hooks/use-assign-licenses-to-site.ts#L97` and add `type: 'WPCOM_HOSTING',` so it looks like

```
return {
	selectedSite: selectedSite?.domain || '',
	selectedProducts: apiRequests,
	type: 'wpcom-hosting',
};
```

4. Click on the `...` on any site > Click `Issue new license` > Select and issue a license > Verify that you are redirected to the dashboard and below notification is shown. Note: This will be shown only when we assign a WPCOM atomic hosting plan which has not yet been added.
5. Verify that both the buttons work as expected and the banner is dismissed when the button is clicked, or the close button is clicked.
6. Verify that appropriate track event calls are being made when these buttons are clicked.
7. Verify this banner looks good and all the different devices.

![mobile (38)](https://github.com/Automattic/wp-calypso/assets/10586875/51f29a6c-bd76-41c2-8e8d-61b91abcb0d8)

----

Mobile

![mobile (36)](https://github.com/Automattic/wp-calypso/assets/10586875/a687ffc0-77d5-4505-8743-b1ae32883cca)

----
Tablet

![mobile (37)](https://github.com/Automattic/wp-calypso/assets/10586875/828bf477-b0ef-4cbb-8f83-c116077fe19f)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Update <Banner /> component to show extraContent if available

yashwin committed:
Implement UI to show WPCOM atomic hosting notifications

yashwin committed:
Update SiteAddLicenseNotification to show WPCOM hosting notification if the type is "wpcom-hosting"

mattgawarecki commented on client/components/banner/index.jsx:
```diff
@@ -246,6 +248,7 @@ export class Banner extends Component {
 							) ) }
 						</ul>
 					) }
+					{ extraContent && <>{ extraContent }</> }
```
Non-blocking:

Is this not the same as `{ extraContent }`? If the value is empty, nothing will be rendered; if non-empty, the value will be rendered as normal.

yashwin replied:
Agree. I have made the changes: https://github.com/Automattic/wp-calypso/pull/81173/commits/a14a2af29983355f19a3d8da28314a35a46fde11

mattgawarecki commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-add-license-notification/wpcom-atomic-hosting-notification.tsx:
```diff
@@ -0,0 +1,88 @@
+			<Button
+				borderless
+				target="_blank"
+				href={ `https://wordpress.com/domains/manage/${ selectedSite }` }
```
Just checking here, since it's a bug that came up recently: is `selectedSite` is a slug here, rather a URL?

yashwin replied:
Oh, yes. Good catch. I have made the changes: https://github.com/Automattic/wp-calypso/pull/81173/commits/a14a2af29983355f19a3d8da28314a35a46fde11

mattgawarecki commented:
@yashwin I followed the test instructions but wasn't able to see the "Congratulations!" banner at step 4. See the attached video:

https://github.com/Automattic/wp-calypso/assets/670067/11790548-144f-453e-97b6-2a270d2d8e92

yashwin committed:
Address PR comments

yashwin commented:
@mattgawarecki: Thanks for the thorough review. I have addressed all your comments.

> I followed the test instructions but wasn't able to see the "Congratulations!" banner at step 4.

I'm sorry. I changed WPCOM_HOSTING to be a constant with the value `wpcom-hosting` and forgot to change the testing instruction. I have updated it now.

yashwin commented:
@ilonagl: Thanks for the review. 

> It would be great if you could align the checkbox and the close icon with the top header (image below).

I wanted to keep it aligned with all the banners. Anyway, I have updated it now.

> Can we tweak the primary green button paddings, by making the top margins 8px? The current one that we have in our library looks big within the JITM. What do you think?

Yeah, agree! It was already 8px. I have now reduced it to 4px, and it looks better now.

<img width="1613" alt="Screenshot 2023-09-01 at 10 14 45 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/e7cc0cdd-62d1-418a-8079-0a3b9275194c">

<img width="376" alt="Screenshot 2023-09-01 at 10 36 10 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/99a2c1f0-b277-46ab-a454-156a0436e4bb">

jkguidaven reviewed (approved):
working as expected as per test instruction. code LGTM 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8868625

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

emilyaudela commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-add-license-notification/wpcom-atomic-hosting-notification.tsx:
```diff
@@ -0,0 +1,91 @@
+					handleOnClick( 'calypso_jetpack_agency_dashboard_setup_site_in_wp_admin_click' )
+				}
+			>
+				{ translate( 'Setup site in wp-admin' ) }
```
Small typo: `Setup` --> `Set up` (Since it's a verb here and not an adjective)

yashwin replied:
@emilyaudela: Thanks for bringing this up. This is fixed here: https://github.com/Automattic/wp-calypso/pull/81495
